frontend/test/metabase/scenarios/models/model-actions.cy.spec.js

```diff
@@ -0,0 +1,201 @@
+      cy.findByText("Create").should("be.visible");
+      cy.findByText("Update").should("be.visible");
+      cy.findByText("Delete").should("be.visible");
+    });
+
+    cy.findByRole("button", { name: "New action" }).click();
+    fillQuery("DELETE FROM orders WHERE id = {{ id }}");
+    fieldSettings()
+      .findByText(/Number/i)
+      .click();
+    cy.findByRole("button", { name: "Save" }).click();
+    modal().within(() => {
+      cy.findByLabelText("Name").type("Delete Order");
+      cy.findByRole("button", { name: "Create" }).click();
+    });
+    cy.findByLabelText("Action list")
+      .findByText("Delete Order")
+      .should("be.visible");
+
+    openActionEditorFor("Delete Order");
+    fillQuery(" AND status = 'pending'");
+    fieldSettings()
+      .findByRole("radiogroup", { name: /Field type/i })
+      .findByLabelText(/Number/i)
+      .should("be.checked");
+    cy.findByRole("button", { name: "Update" }).click();
+
+    cy.findByLabelText("Action list")
+      .findByText(
+        "DELETE FROM orders WHERE id = {{ id }} AND status = 'pending'",
+      )
+      .should("be.visible");
+  });
+
+  it("should allow to make actions public and execute them", () => {
+    cy.intercept("/api/public/action/*/execute", request => {
+      expect(request.body).to.deep.equal({
+        parameters: { [TEST_PARAMETER.id]: -2 },
+      });
+    });
+
+    cy.get("@modelId").then(modelId => {
+      cy.request("POST", "/api/action", {
+        ...SAMPLE_QUERY_ACTION,
+        model_id: modelId,
+      });
+      cy.visit(`/model/${modelId}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+    openActionEditorFor(SAMPLE_QUERY_ACTION.name);
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("button", { name: "Action settings" }).click();
+      cy.findByLabelText("Make public").should("not.be.checked").click();
+      cy.findByLabelText("Public action link URL")
+        .invoke("val")
+        .then(url => {
+          cy.wrap(url).as("publicUrl");
+        });
+    });
+
+    cy.get("@publicUrl").then(url => {
+      cy.signOut();
+      cy.visit(url);
+    });
+
+    cy.findByLabelText(TEST_PARAMETER.name).type("-2");
+    cy.findByRole("button", { name: "Submit" }).click();
+    cy.findByText("Thanks for your submission.").should("be.visible");
+    cy.findByRole("form").should("not.exist");
+    cy.findByRole("button", { name: "Submit" }).should("not.exist");
+
+    cy.signInAsAdmin();
+    cy.get("@modelId").then(modelId => {
+      cy.visit(`/model/${modelId}/detail`);
+      cy.wait("@getModel");
+    });
+
+    cy.findByText("Actions").click();
+    openActionEditorFor(SAMPLE_QUERY_ACTION.name);
+
+    cy.findByRole("dialog").within(() => {
+      cy.findByRole("button", { name: "Action settings" }).click();
+      cy.findByLabelText("Make public").should("be.checked").click();
+    });
+    modal().findByRole("button", { name: "Yes" }).click();
```

Comment:
I'd add another assertion within the modal that checks if the header **Disable this public link?** exists. This would help make it more obvious that we're trying to disable/delete the public link. Because it could be ambiguous what **Yes** button we're clicking here.
<img width="688" alt="image" src="https://user-images.githubusercontent.com/1937582/215200342-6d201edb-5277-42b1-8f7c-445dbe89b136.png">


What do you think?